zulipterminal/ui_tools/views.py

```diff
@@ -1169,6 +1169,39 @@ def __init__(self, controller: Any, title: str) -> None:
         super().__init__(controller, widgets, "HELP", popup_width, title)
 
 
+class FileUploadView(PopUpView):
+    def __init__(
+        self,
+        controller: Any,
+        write_box: WriteBox,
+        title: str,
+    ) -> None:
+        self.controller = controller
+        self.model = controller.model
+        self.write_box = write_box
+        max_cols, max_rows = controller.maximum_popup_dimensions()
+        self.predefined_text = urwid.Text("Location : ")
+        self.file_location_edit = urwid.Edit()
+        columns = [self.predefined_text, self.file_location_edit]
+        super().__init__(
+            controller,
+            columns,
+            "FILE_UPLOAD",
+            max_cols,
+            title,
+        )
+
+    def _handle_file_upload(self, file_location: str) -> None:
+        self.uri = self.model.get_file_upld_uri(file_location)
+        file_name = file_location.split("/")[-1]
```

Comment:
This works for now, but I think in case of Windows, the file path contains [backward slashes](https://learn.microsoft.com/en-us/dotnet/standard/io/file-path-formats), and this approach may not work there.

Reply:
Here to get the `file_name` we are using the file location and NOT the `url`. So not sure how `urllib` can be used directly to get the file name.